### PR TITLE
[CSM-146] Fix api docs.

### PR DESCRIPTION
**Pull request by ksaric**
Working docs generation, added `docs` folder for the destination.

### CODE REVIEW

File: src/Pos/Wallet/Web/Doc.hs [hunk 1]
```diff
@@ -231,28 +305,49 @@ instance ToSample CAddress where
...
 
 instance ToSample BackupPhrase where
-    toSamples Proxy = error "unmaintained stub for [ToSample:BackupPhrase]"
+    toSamples Proxy = singleSample sample
+      where
+        sample = backupPhrase
 
 instance ToSample SoftwareVersion where
-    toSamples Proxy = singleSample curSoftwareVersion
+      toSamples Proxy = singleSample curSoftwareVersion
```
Comment (akegalj) [inline src/Pos/Wallet/Web/Doc.hs]: minor, formatting (extra space)

File: src/Pos/Wallet/Web/Doc.hs [hunk 2]
```diff
@@ -206,23 +242,61 @@ instance ToSample Coin where
 --
 -- FIXME!
 instance ToSample CHash where
-    toSamples Proxy = error "unmaintained stub for [ToSample:CHash]"
+    toSamples Proxy = singleSample $ CHash "1fi9sA3pRt8bKVibdun57iyWG9VsWZscgQigSik6RHoF5Mv"
```
Comment (akegalj) [inline src/Pos/Wallet/Web/Doc.hs]: you can probably add to your editor to remove extra witespace at the end of the line

Comment (ksaric) [inline src/Pos/Wallet/Web/Doc.hs]: I think It is set so it removes extra whitespace?

File: src/Pos/Wallet/Web/Doc.hs (hunk 1 above)
Comment (ksaric) [inline src/Pos/Wallet/Web/Doc.hs]: Fixed.

File: docs/wallet-web-api.md [hunk 3]
```diff
@@ -0,0 +1,1518 @@
...
+
+    - `application/json`
+
+- Example: `application/json`
+
+```javascript
+{
+    "cwBackupPhrase": {
+        "bpToList": [
+            "me",
```
Comment (rockbmb) [inline docs/wallet-web-api.md]: What's "me" supposed to mean here?

File: src/Pos/Wallet/Web/Doc.hs [hunk 4]
```diff
@@ -1,4 +1,6 @@
 {-# LANGUAGE DataKinds #-}
+{-# LANGUAGE OverloadedStrings #-}
```
Comment (gromakovsky) [inline src/Pos/Wallet/Web/Doc.hs]: It's enabled by default.

File: src/Pos/Wallet/Web/Doc.hs [hunk 5]
```diff
@@ -1,4 +1,6 @@
...
+{-# OPTIONS_GHC -fno-warn-deprecations #-}
```
Comment (gromakovsky) [inline src/Pos/Wallet/Web/Doc.hs]: If everything is implemented, do we need this suppression?

File: src/wallet-web-docs/Main.hs [hunk 6]
```diff
@@ -3,6 +3,7 @@ import           Universum
 import           Pos.Wallet.Web (walletDocsText)
 
 main :: IO ()
-main = writeFile fp walletDocsText >> putStrLn ("See " <> fp)
-  where
-    fp = "wallet-web-api.md"
+main = do
```
Comment (gromakovsky) [inline src/wallet-web-docs/Main.hs]: Looks redundant.

File: docs/wallet-web-api.md (hunk 3 above)
Comment (ksaric) [inline docs/wallet-web-api.md]: Nothing, really. It's supposed to be a funny reference, seems it backfired on me - `mkBackupPhrase $ replicate 12 "me"`. Do you need me to change this to something else?

Comment (rockbmb) [inline docs/wallet-web-api.md]: It was unclear to me that this was just a placeholder. Can you pick 12 random words from the dictionary we use to sample backup words from instead?

Comment (ksaric) [inline docs/wallet-web-api.md]: Sure thing. Do you have some example that you prefer or should I use any words I find? For example, here - https://iancoleman.github.io/bip39/?

Comment (rockbmb) [inline docs/wallet-web-api.md]: Well, as long as it's an allowed sequence of words, i.e. all of them are permitted backup phrase words, any example is fine. You can repeat the same 12 words throughout this file and the entire project, that's fine. Just pick 12 random ones once.

Comment (ksaric) [inline docs/wallet-web-api.md]: Ok. Fixed. Used words are `transfer uniform grunt excess six veteran vintage warm confirm vote nephew allow`.

File: src/Pos/Wallet/Web/Doc.hs (hunk 4 above)
Comment (ksaric) [inline src/Pos/Wallet/Web/Doc.hs]: Ok. Fixed.

File: src/Pos/Wallet/Web/Doc.hs (hunk 5 above)
Comment (ksaric) [inline src/Pos/Wallet/Web/Doc.hs]: I guess not. For now. Fixed.

File: src/wallet-web-docs/Main.hs (hunk 6 above)
Comment (ksaric) [inline src/wallet-web-docs/Main.hs]: I left this since I was thinking of adding another generated document, but I can fix it for now. Fixed.

**Comment by akegalj** [comment]
Are we sure we want to commit produced wallet-web docs to the repo?

**Comment by rockbmb** [comment]
If it's under the `/technical` section, I don't see a problem with that.

**Comment by gromakovsky** [comment]
What is `/technical` section?

I think that if we want to have produced doc in repo, we should do something to update it automatically. Otherwise it will often be outdated.

**Comment by rockbmb** [comment]
@gromakovsky Whoops. I haven't slept in I don't know how many hours, I thought this was `cardano-docs.iohk`. @akegalj then raises a fair point. As such, this should be moved to documentation repo IMO, and then my previous comment will be valid.

**Comment by ChShersh** [comment]
:+1: I agree that we should do something with automatic documentation update and put some effort in it. But still almost nobody cares...

**Comment by rockbmb** [comment]
@ChShersh I'm giving it some thought presently, these are good suggestions. Maybe someone knows of some tool for this.

**Pull request merged**